crates/fuzzing/src/generators/table_ops.rs

```diff
@@ -44,31 +48,50 @@ impl TableOps {
     /// calls), but is not guaranteed to avoid traps (might access out-of-bounds
     /// of the table).
     pub fn to_wat_string(&self) -> String {
-        let mut wat = "(module\n".to_string();
+        let mut module = Module::new();
 
         // Import the GC function.
-        wat.push_str("  (import \"\" \"gc\" (func))\n");
+        let mut imports = ImportSection::new();
+        imports.import("", "gc", ImportType::Function(0));
 
         // Define our table.
-        wat.push_str("  (table $table ");
-        write!(&mut wat, "{}", self.table_size()).unwrap();
-        wat.push_str(" externref)\n");
+        let mut tables = TableSection::new();
+        tables.table(TableType {
+            element_type: ValType::ExternRef,
+            limits: Limits {
+                min: self.table_size(),
+                max: None,
+            },
+        });
 
         // Define the "run" function export.
-        wat.push_str(r#"  (func (export "run") (param"#);
-        for _ in 0..self.num_params() {
-            wat.push_str(" externref");
+        let mut functions = FunctionSection::new();
+        functions.function(0);
+
+        let mut exports = ExportSection::new();
+        exports.export("run", Export::Function(0));
+
+        let mut params: Vec<(u32, ValType)> = Vec::with_capacity(self.num_params() as usize);
+        for i in 0..self.num_params() {
+            params.push((0, ValType::ExternRef));
         }
-        wat.push_str(")\n");
+        let mut func = Function::new(params);
+
         for op in self.ops.iter().take(MAX_OPS) {
-            wat.push_str("    ");
-            op.to_wat_string(&mut wat);
-            wat.push('\n');
+            func.instruction(op.to_instruction());
         }
-        wat.push_str("  )\n");
 
-        wat.push_str(")\n");
-        wat
+        let mut code = CodeSection::new();
+        code.function(&func);
+
+        module
+            .section(&imports)
+            .section(&functions)
+            .section(&tables)
+            .section(&exports)
+            .section(&code);
+
+        String::from_utf8(module.finish()).unwrap() // TODO: Doesn't actually turn into wat string
```

Comment:
The return type of this function will probably want  to change to `Vec<u8>` since a string is no longer constructed.